python/lsst/daf/butler/cli/opt/repo.py

```diff
@@ -23,13 +23,34 @@
 import click
 
 
-class repo_option:  # noqa: N801
-    def __init__(self, required=False, help=None):
-        self.help = help if help is not None else """Path to an existing data repository root or configuration
-                    file."""
+class repo_argument:  # noqa: N801
+    """Decorator to add a repo argument to a click command.
+
+    Parameters
+    ----------
+    required : bool, optional
+        Indicates if the caller must pass this argument to the command, by
+        default True.
+    help : str, optional
+        The help text for this argument to append to the command's help text.
+        If None or '' then nothing will be appended to the help text (in which
+        case the command should document this argument directly in its help
+        text). By default the value of repo_argument.existing_repo.
+    """
+
+    will_create_repo = "REPO is a filesystem path for the new repository. " \
+                       "Will be created if it does not exist."
+    existing_repo = "REPO is the path to an existing data repository root " \
```

Comment:
This is a URI to an existing repository root or configuration file. Does not have to be a path.